### DISCUSSION TITLE
Markup semantics

### DISCUSSION THREAD

**Discussion by MasterInQuestion** [MasterInQuestion/talk#28] [Linguistic]
= Markdown =

https://github.com/orgs/community/discussions/20921
[[
<b id="issuecomment-1188472508">[ MasterInQuestion @ CE 2022-07-19 00:42:12 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188472508](https://github.com/curl/curl/issues/9162#issuecomment-1188472508)</b>
    There's semantic difference between "<code>&#x60;</code>" (Grave Accent) (U+0060, 0x60, &amp;<span>#</span>96;) and &#x60;`"`&#x60; (Quotation Mark) (U+0022, 0x22, &amp;<span>#</span>34;).
    .
    E.g.
    |\*| &#x60;`\x7F`&#x60; means:
(Character Sequence)
[[
|1| &#x60;`\`&#x60; (Reverse Solidus) (U+005C, 0x5C, &amp;<span>#</span>92;)
|2| "`x`" (Latin Small Letter X) (U+0078, 0x78, &amp;<span>#</span>120;)
|3| "`7`" (Digit 7) (U+0037, 0x37, &amp;<span>#</span>55;)
|4| "`F`" (Latin Capital Letter F) (U+0046, 0x46, &amp;<span>#</span>70;)
]]
    |\*| "`\x7F`" means "`\x7F`" (Delete) (U+007F, 0x7F, &amp;<span>#</span>127;). <b>]</b>

\----

<b id="issuecomment-1188609745">[ vszakats @ CE 2022-07-19 05:19:25 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188609745](https://github.com/curl/curl/issues/9162#issuecomment-1188609745)</b>
    It's a Markdown markup here, meaning to display it as code. <b>]</b>

\----

<b id="issuecomment-1188622845">[ MasterInQuestion @ CE 2022-07-19 05:42:57 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188622845](https://github.com/curl/curl/issues/9162#issuecomment-1188622845)</b>
    I know. And such sophisticated markup creates general interoperability issue.
    See [ [ https://github.com/curl/curl/issues/9153#issuecomment-1185051870](https://github.com/curl/curl/issues/9153#issuecomment-1185051870) ]. <b>]</b>

\----

<b id="issuecomment-1188627816">[ vszakats @ CE 2022-07-19 05:51:26 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188627816](https://github.com/curl/curl/issues/9162#issuecomment-1188627816)</b>
    Speaking of GitHub Issues, I think it's the opposite. <b>]</b>

\----

<b id="issuecomment-1188636804">[ MasterInQuestion @ CE 2022-07-19 06:07:34 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188636804](https://github.com/curl/curl/issues/9162#issuecomment-1188636804)</b>
    For GitHub yet has broken rendering on some very fundamental thing.
    .
    What's the purpose of all these sophisticated markups if things may be very well clearly expressed in Plain Text? <b>]</b>

\----

<b id="issuecomment-1188707772">[ vszakats @ CE 2022-07-19 07:39:06 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188707772](https://github.com/curl/curl/issues/9162#issuecomment-1188707772)</b>
    The purpose of it is to make posts easier to read and harder to misunderstand. For example code markup will retain spaces and other things.
    It also indicates that it is code/command/input/output, or a &#x60;keyword&#x60;. Yet it remains readable as plain text as well. <b>]</b>

\----

<b id="issuecomment-1188726185">[ MasterInQuestion @ CE 2022-07-19 07:58:22 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188726185](https://github.com/curl/curl/issues/9162#issuecomment-1188726185)</b>
    The crucial problem is that such significant information become lost when copying (from the rendered display) back into Plain Text.
    Which made general parsing / exchanging bothersome. <b>]</b>

\----

<b id="issuecomment-1188739624">[ vszakats @ CE 2022-07-19 08:12:06 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1188739624](https://github.com/curl/curl/issues/9162#issuecomment-1188739624)</b>
    You can always access the original text for your own submissions.
    For others, you can do the same with the "Quote reply" option, if that's necessary. <b>]</b>

\----

<b id="issuecomment-1189590578">[ MasterInQuestion @ CE 2022-07-19 21:57:32 UTC:
[ https://github.com/curl/curl/issues/9162#issuecomment-1189590578](https://github.com/curl/curl/issues/9162#issuecomment-1189590578)</b>
    Not always (e.g. locked topic). And the practice is bothersome (not only login required, and the apparent more).
    Even if so:

    Readable in one way, unreadable in another:
    The doomed fate of Markdown, and similar sophisticated markups:
    That demand sophisticated post-rendering to be properly displayed. <b>]</b>

\----

<b id="issuecomment-1192454350">[ jzakrzewski @ CE 2022-07-22 11:01:44 UTC:
[ https://github.com/curl/curl/issues/9189#issuecomment-1192454350](https://github.com/curl/curl/issues/9189#issuecomment-1192454350)</b>
    "Your editing to the main post screwed up the formatting seriously (and caused misinformation). I reverted your change for readability. [Note 1]"
<^>    Sorry, but I find your post completely unreadable with weird formatting:
    Quoting, notes, jumping around...
    Markdown compared to that is crystal clear. <b>]</b>

\----

<b id="issuecomment-1192456554">[ MasterInQuestion @ CE 2024-05-30 07:16:44 UTC:
[ https://github.com/curl/curl/issues/9189#issuecomment-1192456554](https://github.com/curl/curl/issues/9189#issuecomment-1192456554)</b>
    Further Markdown related Off-Topic to [ [ https://github.com/orgs/community/discussions/20921](https://github.com/orgs/community/discussions/20921) ], thanks.

    ----

    Try think of any way superior maintaining the very structure:
    These elements are unavoidable for effective representation.

    (assuming rendered Markdown)
    This already put the comparison at a biased position.
    With proper post-rendering (just mere syntax highlighting): the schema should be far more accessible than Markdown.
    Without post-rendering: the schema should also maintain high level of accessibility; while Markdown... seems Markdone.

    ----

    I fear the real problem in fact has little to do with Markdown or whatsoever markup scheme. But the general ability of parsing information.
    These very elements may as well present in Markdown, in no way essentially different. <b>]</b>

\----

<b id="issuecomment-1192750136">[ bagder @ CE 2022-07-22 16:35:28 UTC:
[ https://github.com/curl/curl/issues/9189#issuecomment-1192750136](https://github.com/curl/curl/issues/9189#issuecomment-1192750136)</b>
    I'm with @jzakrzewski. All this weirdly formatted mumbo jumbo makes it very hard to extract the actual content that might be in there somewhere.
    So, again why would we want this new option? (please reply without all the funny formatting) <b>]</b>
]]

    Related:
    [ https://github.com/MasterInQuestion/talk/discussions/3#discussioncomment-11521651](https://github.com/MasterInQuestion/talk/discussions/3#discussioncomment-11521651)

**Comment by adlai** [comment]
I think the original purpose of Markdown, and similar formats:
    Was not for the raw source to get processed into HTML; although this was supposed to be easy when desired.

    However these formats (YAML being another one that comes to mind) are supposed to be sufficiently minimal:
    That displaying them raw is still comfortably readable, for both human and machine consumption.

    One easy hack for avoiding the post-processing of Markdown in GitHub comments:
    Is to enclose your entire comment in a code block using triple backticks ("<code>&#x60;&#x60;&#x60;</code>").
    .
    This gives you complete control of layout within the large code block.
    And if you do use Markdown within this code block, then the only pitfall to avoid is nested code blocks:
    Any nested blocks should use indentation, rather than triple backticks.

**Reply by MasterInQuestion** [reply]
To effectively store information, one has to first: devise an appropriate schema.
    .
    Well structured as if formal records while formless like water:
    A powerful data language is crucial for the instantiation of creativity.
\
\
    Such is how alike ends:
    https://github.com/orgs/community/discussions/16929
    (malapropos ["white-space"](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space) handling)

    I did test the various before.
